lib/maxwell/adapter/util.ex

```diff
@@ -136,9 +136,24 @@ defmodule Maxwell.Adapter.Util do
   end
   defp multipart_body(:end), do: :eof
 
-  defp append_query_string(url, path, query)when query == %{}, do: url <> path
+  defp append_query_string(url, path, query)when query == %{}, do: url_joint_path(url, path)
   defp append_query_string(url, path, query) do
     query_string = URI.encode_query(query)
-    url <> path <> "?" <> query_string
+    url_path = url_joint_path(url ,path)
+    url_path <> "?" <> query_string
   end
+
+  defp url_joint_path(url, path) do
```

Comment:
You can consult `Path.join` which has a fantastic implementation of join pathes.
It use patten matching to implement `String.starts_with?` which will definitely be more effecient.